query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestEligibilityValidator.java

```diff
@@ -0,0 +1,148 @@
+          || queryRequest.getDistinctSelections()) {
+        return QueryCost.UNSUPPORTED;
+      }
+
+      Set<String> referencedColumns = executionContext.getReferencedColumns();
+      Preconditions.checkArgument(!referencedColumns.isEmpty());
+      String timeFilterColumn = executionContext.getTimeFilterColumn();
+      // all the columns in the request should have a mapping in the config
+      for (String referencedColumn : referencedColumns) {
+        if (!timeFilterColumn.equals(referencedColumn)
+            && prometheusViewDefinition.getPhysicalColumnName(referencedColumn) == null
+            && prometheusViewDefinition.getMetricConfig(referencedColumn) == null) {
+          return QueryCost.UNSUPPORTED;
+        }
+      }
+
+      if (!analyseAggregationColumns(queryRequest.getAggregationList())) {
+        return QueryCost.UNSUPPORTED;
+      }
+
+      if (!analyseSelectionAndGroupBy(
+          queryRequest.getSelectionList(), queryRequest.getGroupByList())) {
+        return QueryCost.UNSUPPORTED;
+      }
+
+      if (!analyseFilter(queryRequest.getFilter())) {
+        return QueryCost.UNSUPPORTED;
+      }
+    } catch (Exception e) {
+      return QueryCost.UNSUPPORTED;
+    }
+    // value 1.0 so that prometheus is preferred over others
+    return new QueryCost(1.0);
+  }
+
+  private boolean analyseSelectionAndGroupBy(
+      List<Expression> selectionList, List<Expression> groupByList) {
+    Set<String> selections = Sets.newHashSet();
+    for (Expression expression : selectionList) {
+      if (!QueryRequestUtil.isSimpleColumnExpression(expression)) {
+        return false;
+      }
+      selections.add(getLogicalColumnNameForSimpleColumnExpression(expression));
+    }
+
+    for (Expression expression : groupByList) {
+      // skip datetime convert group by
+      if (QueryRequestUtil.isDateTimeFunction(expression)) {
+        continue;
+      }
+      if (!QueryRequestUtil.isSimpleColumnExpression(expression)) {
+        return false;
+      }
+      if (!selections.remove(getLogicalColumnNameForSimpleColumnExpression(expression))) {
+        return false;
+      }
+    }
+    // all selection and group by should be on same column
+    return selections.isEmpty();
+  }
+
+  private boolean analyseAggregationColumns(List<Expression> aggregationList) {
+    for (Expression expression : aggregationList) {
+      Function function = expression.getFunction();
+      // skip dateTimeConvert function as it is part of the prometheus api call
+      if (QueryRequestUtil.isDateTimeFunction(expression)) {
+        continue;
+      }
+      if (function.getArgumentsCount() > 1) {
+        return false;
+      }
+      Expression functionArgument = function.getArgumentsList().get(0);
+      if (!QueryRequestUtil.isSimpleColumnExpression(functionArgument)) {
+        return false;
+      }
+      String attributeName = getLogicalColumnNameForSimpleColumnExpression(functionArgument);
```

Comment:
attributeId (name is something like "key", id, which we use as col names are of the format "<SCOPE>.key").